openedx/features/colaraz_features/forms.py

```diff
@@ -316,53 +268,40 @@ def create(self):
         """
-        roles = bulk_create_course_access_role(
+        created_roles = bulk_create_course_access_role(
             self.user,
             user_id=self.cleaned_data['user'],
             org=self.cleaned_data['org'],
             roles=self.cleaned_data.get('roles', []),
             course_ids=self.cleaned_data.get('course_ids', [])
         )
-
-        return roles[0] if len(roles) > 0 else None
+        return created_roles[0] if len(created_roles) > 0 else None
 
     def update(self):
         """
         Update course access roles, deleting the ones removed by the user and return one instance.
         """
-        # if user or org has changed then delete all the roles that match self.instances and create new ones.
-        if str(self.instance.user.id) != self.cleaned_data['user'] or self.instance.org != self.cleaned_data['org']:
-            self.instances.delete()
-            return self.create()
-
-        instance = None
-        if not self.instance.course_id:
-            # We are on the organization level course access role management page.
-            self.instances.exclude(
-                role__in=self.cleaned_data.get('roles', [])
-            ).delete()
-
-            for role in self.cleaned_data.get('roles', []):
-                instance = get_or_create_course_access_role(
-                    self.user,
-                    user_id=self.cleaned_data['user'],
-                    org=self.cleaned_data['org'],
-                    role=role,
-                )
-        else:
-            # We are on course level access roles page.
-            self.instances.exclude(
-                role__in=self.cleaned_data.get('roles', []),
-                course_id__in=self.cleaned_data.get('course_ids', [])
-            ).delete()
-
-            for course_id in self.cleaned_data.get('course_ids', []):
-                for role in self.cleaned_data.get('roles', []):
-                    instance = get_or_create_course_access_role(
-                        self.user,
-                        user_id=self.cleaned_data['user'],
-                        org=self.cleaned_data['org'],
-                        role=role,
-                        course_id=course_id,
-                    )
-        return instance
+        def _is_course_creator_role_changed(initial_state, final_state):
+            return CourseCreatorRole.ROLE in initial_state.get('roles', []) \
+                and CourseCreatorRole.ROLE not in final_state.get('roles', [])
+
+        if _is_course_creator_role_changed(self.initial, self.cleaned_data):
+            revoke_course_creator_access(self.instance.user, self.user)
+            self.instances.filter(role=CourseCreatorRole.ROLE).delete()
+
+        cleaned_roles = []
+        for role in self.cleaned_data.get('roles', []):
+            cleaned_roles.append(role.replace('org_', '', 1) if role.startswith('org_') else role)
```

Comment:
```suggestion
            cleaned_roles.append(role.replace('org_', '', 1))
```
do we need this if else check?

Reply:
It makes sure that the 'org_' keyword is in the start, so later on if a role has org_ anywhere else it will not create any unexpected behavior.